File: 240129/Carry 피하기 2/escaping-carry-2.py
n=int(input())

num=[int(input()) for _ in range(n)]

def carry(x,y):
    x=list(str(x))
    y=list(str(y))
    x=x[::-1]
    y=y[::-1]
    if len(x)<=len(y):
        for i in range(len(x)):
            if min(int(x[i]),int(y[i]))>(int(x[i])+int(y[i]))%10:
                return False
    else:
        for i in range(len(y)):
            if min(int(x[i]),int(y[i]))>(int(x[i])+int(y[i]))%10:
                return False
    return True

def add(x,y,z):
    if carry(x,y) and carry(x+y,z):
        return x+y+z
    else:
        return 0
ans=0
for i in range(n-2):
    for j in range(i+1,n-1):
        for k in range(j+1,n):
            ans=max(ans,add(num[i],num[j],num[k]))

if ans==0:
    print(-1)
else:
    print(ans)

File: 260106/Carry 피하기 2/escaping-carry-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const arr = input.slice(1, 1 + n);

// Please Write your code here.
function carry(arr){
    const nums = Array.from({length:n}, ()=>Array(0));
    for(let i=0; i<arr.length; i++){
        for(let j=0; j<arr[i].length; j++){
            nums[j].push(Number(arr[i][arr[i].length-j-1]))
        }

    }

    for(let i=0; i<nums.length; i++){
        let sum = 0;
        for(let j=0; j<nums[i].length; j++){
            sum+=nums[i][j];
        }
        if(sum>=10){
            return -1;
        }
    }

    let sum = 0;
    for(let i=0; i<arr.length; i++){
        sum+=Number(arr[i]);
    }
    return sum;
}


let ans = -1;

for(let i=0; i<n; i++){
    for(let j=i+1; j<n; j++){
        for(let k=j+1; k<n; k++){
            ans = Math.max(ans, carry([arr[i], arr[j], arr[k]]))
        }
    }
}

console.log(ans)
